test/e2e/network/network_policy.go

```diff
@@ -353,6 +402,197 @@ var _ = SIGDescribe("NetworkPolicy", func() {
 				testCannotConnect(f, f.Namespace, clientPodName, service, 81)
 			})
 		})
+
+		ginkgo.It("should enforce updated policy [Feature:NetworkPolicy]", func() {
+			const (
+				clientAAllowedPort    = 80
+				clientANotAllowedPort = 81
+			)
+			ginkgo.By("Creating a network policy for the Service which allows traffic from pod at a port")
+			policy := &networkingv1.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "allow-ingress",
+				},
+				Spec: networkingv1.NetworkPolicySpec{
+					// Apply to server
+					PodSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"pod-name": podServer.Name,
+						},
+					},
+					// Allow traffic only to one port.
+					Ingress: []networkingv1.NetworkPolicyIngressRule{{
+						From: []networkingv1.NetworkPolicyPeer{{
+							PodSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									"pod-name": "client-a",
+								},
+							},
+						}},
+						Ports: []networkingv1.NetworkPolicyPort{{
+							Port: &intstr.IntOrString{IntVal: clientAAllowedPort},
+						}},
+					}},
+				},
+			}
+			policy, err := f.ClientSet.NetworkingV1().NetworkPolicies(f.Namespace.Name).Create(policy)
+			framework.ExpectNoError(err, "Error creating Network Policy %v: %v", policy.ObjectMeta.Name, err)
+
+			testCanConnect(f, f.Namespace, "client-a", service, clientAAllowedPort)
+			err = f.WaitForPodNotFound("client-a", framework.PodDeleteTimeout)
```

Comment:
Would it be faster to use unique names, rather than waiting for the teardown to reuse the name? (I suspect so)

Reply:
Yes, it would be faster, but it would not cover the corner cases generated when updating a policy rule to use a different `podSelector`. The name given to a pod is also used for labeling it. Thus, I believe is good to ensure that, for example, when the `podSelector` on a `NetworkPolicyIngressRule` is updated the pod that was previous selected is not selected anymore.

Reply:
I see. A bit unfortunate, but not worth blocking on.